### DISCUSSION TITLE
Please add custom keyboard click sounds as a setting

### DISCUSSION THREAD

**Discussion by mak448a** [monkeytypegame/monkeytype#2134] [Ideas] [answered]
Could you add the ability to use your own keyboard click sounds from your documents in the setting menu? If you can that would be great. Thanks.

**Comment by devkennyy** [comment]
See [Mechvibes](https://github.com/hainguyents13/mechvibes/releases/tag/v2.1.1).

**Reply by wizard-waqas** [reply]
This isn't a solution, its a workaround. This would be like if the speakers on my computer didn't work, so I watched the video on my computer screen and played the audio from my phone. See [KeyboardSim](https://github.com/tplai/kbsim). We could implement sounds in browser so the user would not have to download anything additional

**Reply by mak448a** [reply]
This doesn't work 100% on Linux under Wayland

**Comment by unrealapex** [comment]
Although this question has already been answered, please make sure to check if your question has already been asked by someone else. This is a duplicate of #2619.

**Comment by Miodec** [comment, marked as answer]
Real answer why this isnt a feature: Handling file uploads is dangerous and difficult. 

There would need to be a way to do this only locally.

**Reply by Juliaria08** [reply]
Would it be possible to fetch a sound file from a url on the client side?, (I hope it'd allow having 127.0.0.1, but else I'd mount it on a remote server (but I know everyone doesn't have that luxuary))

**Reply by dsinkerii** [reply]
would be cool to see something like a support for [Myinstants](https://www.myinstants.com/en/index/us/), so you can either paste in the id of the sound, or a link of it. but if the current structure of the code doesnt allow any form of custom sounds, then perhaps add more sounds suggested by people?

**Comment by shardulvs** [comment]
I wanted a machine gun sound
